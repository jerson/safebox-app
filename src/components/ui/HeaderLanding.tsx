import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  View,
  ViewStyle,
  Image,
  ImageStyle,
  TextStyle,
  StyleProp
} from 'react-native';
import Colors from '../../modules/constants/Colors';
import { NavigationScreenComponent } from 'react-navigation';
import Text from './Text';

const styles = StyleSheet.create({
  logo: {
    width: 90,
    height: 90
  } as ImageStyle,
  logoContent: {
    backgroundColor: Colors.grey6,
    borderRadius: 22,
    borderWidth: 1,
    borderColor: Colors.accent,
    alignSelf: 'center',
    overflow: 'hidden'
  } as ViewStyle,
  logoContainer: {} as ViewStyle,
  header: {
    overflow: 'visible',
    alignItems: 'center',
    justifyContent: 'center'
  } as ViewStyle,
  shadow: {
    shadowOpacity: 0.1,
    shadowRadius: 16,
    shadowColor: Colors.grey6,
    shadowOffset: { height: 4, width: 0 },
    elevation: 4
  } as ViewStyle,
  name: {
    fontSize: 35,
    color: Colors.white
  } as TextStyle,
  subtitle: {
    fontSize: 14,
    color: Colors.grey2
  } as TextStyle
});

export interface HeaderLandingProps {
  subtitle?: string;
  shadow?: boolean;
  style?: StyleProp<ViewStyle>;
  titleStyle?: StyleProp<TextStyle>;
  subtitleStyle?: StyleProp<TextStyle>;
}
const HeaderLanding: React.FC<HeaderLandingProps> = ({
  style,
  subtitle,
  titleStyle,
  subtitleStyle,
  shadow
}) => {
  return (
    <View style={[styles.header, shadow && styles.shadow, style]}>
      <View style={styles.logoContainer}>
        <View style={styles.logoContent}>
          <Image
            style={styles.logo}
            resizeMode={'center'}
            resizeMethod={'scale'}
            source={require('../../assets/images/logo.png')}
          />
        </View>
      </View>

      <Text style={[styles.name, titleStyle]}>
        Safe
        <Text weight={'Bold'}>Box</Text>
      </Text>
      <Text weight={'Light'} style={[styles.subtitle, subtitleStyle]}>
        {subtitle}
      </Text>
    </View>
  );
};

HeaderLanding.defaultProps = {
  subtitle: 'passwords always safe',
  shadow: false
};

export default HeaderLanding;
