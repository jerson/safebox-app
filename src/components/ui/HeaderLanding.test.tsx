import React from 'react';
import * as renderer from 'react-test-renderer';
import HeaderLanding, {HeaderLandingProps} from './HeaderLanding';

const props: HeaderLandingProps = {
  subtitle: 'sample',
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<HeaderLanding {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<HeaderLanding {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
