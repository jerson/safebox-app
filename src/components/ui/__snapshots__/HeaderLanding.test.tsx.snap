// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "overflow": "visible",
      },
      false,
      undefined,
    ]
  }
>
  <_class
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#38414e",
        "borderColor": "#c2d0e7",
        "borderRadius": 22,
        "borderWidth": 1,
        "overflow": "hidden",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={
        Object {
          "testUri": "../../../src/assets/images/logo.png",
        }
      }
      style={
        Object {
          "height": 90,
          "width": 90,
        }
      }
    />
  </_class>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "400",
        },
        Object {},
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 35,
          },
          undefined,
        ],
      ]
    }
  >
    Safe
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "700",
          },
          Object {},
          undefined,
        ]
      }
    >
      Box
    </Text>
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "300",
        },
        Object {},
        Array [
          Object {
            "color": "#f2f4f9",
            "fontSize": 14,
          },
          undefined,
        ],
      ]
    }
  >
    sample
  </Text>
</_class>
`;
